web/includes/functions.php

```diff
@@ -94,6 +94,25 @@ function noCacheHeaders()
     header("Pragma: no-cache");         // HTTP/1.0
 }
 
+function CORSHeaders() {
+	if ( isset( $_SERVER['HTTP_ORIGIN'] ) ) {
+
+		# The following is left for future reference/use.
+		$valid = false;
+		foreach( dbFetchAll( 'SELECT * FROM Servers' ) as $row ) {
+			$Server = new Server( $row );
+			if ( $_SERVER['HTTP_ORIGIN'] == $Server->Url() ) {
+				$valid = true;
+				header("Access-Control-Allow-Origin: " . $Server->Url() );
+				header("Access-Control-Allow-Headers: x-requested-with,x-request");
```

Comment:
Do we need multiple (looks like duplicate) `Access-Control-Allow-Headers` lines, or only a single one?
The `Access-Control-Allow-Origin` per-server makes sense, but I don't know that we need a matching `Access-Control-Allow-Headers` for each one.  Eh?